rclcpp/include/rclcpp/executor.hpp

```diff
@@ -544,7 +544,7 @@ class Executor
   RCLCPP_DISABLE_COPY(Executor)
 
   RCLCPP_PUBLIC
-  void
+  virtual void
```

Comment:
why do you need to make this method virtual?

Reply:
is it because of `spin_until_future_complete`?

IMO having this method as virtual is a bit confusing, but it might be fine

Reply:
I'm still confused if [this](https://github.com/ros2/rclcpp/blob/7bfda87d32bab7e9caa843b2ededa18e427fd00c/rclcpp/include/rclcpp/executor.hpp#L359) will work as expected.

Reply:
I didn't see that `spin_until_future_complete` was being overloaded in the derived class though.
But making `spin_once_impl` virtual doesn't seem to make much harm (though I thing the current code is getting very confusing).